synfig-studio/src/gui/widgets/widget_timetrack.cpp

```diff
@@ -198,6 +198,44 @@ bool Widget_Timetrack::move_selected(synfig::Time delta_time)
 	return ok;
 }
 
+ void Widget_Timetrack::interpolate_selected(synfig::Interpolation type)
+{
+	synfig::Waypoint::Model model;
+	model.set_before(type);
+	model.set_after(type);
+	std::vector<WaypointItem*> selection = waypoint_sd.get_selected_items(); //storing selected waypoint items
+
+	for(int i =0 ; i<selection.size() ; i++) //iteratre through selected waypoint items to fetch the set and apply the interpolation type
+    {
+		std::set<synfig::Waypoint, std::less<synfig::UniqueID> > waypoint_set_new;
+		fetch_waypoints(*selection[i], waypoint_set_new);
+		std::set<synfig::Waypoint, std::less<synfig::UniqueID> >::const_iterator iter;
+
+		for(iter=waypoint_set_new.begin();iter!=waypoint_set_new.end();++iter)
```

Comment:
Prefer [Range-based for loop](https://www.geeksforgeeks.org/range-based-loop-c/).

Reply:
done, although I'm not sure if replacing this const_iter by what I did to use a range-based for loop is safe, so I'm gonna leave this un-resolved just to be sure.